### DISCUSSION TITLE
Romandan Tam Sayıya

### DISCUSSION THREAD

**Discussion by adasue** [sonsuzus/sonsuzus#362] [Soru Cevap]
_Roman rakamları 7 farklı sembol ile gösterilir I, V, X, L, C, D ve M._
**Sembol        Değer**
I                      1
V                     5
X                    10
L                     50
C                   100
D                   500
M                 1000
Örneğin 2 roman rakamı olarak II şeklinde birbirine eklenmiş iki bir yazılır. 12 ise XII diye yazılır, basitçe X + II. 27, XXVII şeklinde yazılır diğer şekilde XX + V + II. 

Roman rakamları genellikle en genişten en küçüğe doğru soldan sağa yazılır. Fakat 4 numarası IIII şeklinde değil. Onun yerine numara 4, IV şeklinde yazılır. Çünkü beşten önceki biri kendisinden çıkararak dört elde ediyoruz. Benzer prensip dokuz için de geçerlidir, IX şeklinde yazılan. Çıkarmanın kullanıldığı örnekler vardır:
I, V ve X öncesine yerleştirilerek 4 ve 9 yapılabilir.
X, L ve C öncesine yerleştirilerek 40 ve 90 yapılabilir.
C, D ve M öncesine yerleştirilerek 400 ve 900 yapılabilir.
Verilen roman rakamını, tam sayıya dönüştürün.

**_Örnek 1:_**
Giriş: s = ”III”
Çıktı: 3
Açıklama: III = 3.

**_Örnek 2:_** 
Giriş: s = “LVIII”
Çıktı: 58
Açıklama: L = 50, V = 5, III = 3  

**_Örnek 3:_**
Giriş: s = “MCMXCIV”
Çıktı: 1994
Açıklama: M = 1000, CM = 900, XC = 90

**_Kısıtlamalar:_**
1 <= s.length <= 15
S sadece ('I', 'V', 'X', 'L', 'C', 'D', 'M') karakterlerini içerir.
S’nin [1, 3999] alanında olan geçerli bir roman numarası olduğu garantilidir.

Kaynakça: https://leetcode.com/problems/roman-to-integer/

**Comment by nilbuyukkinaci** [comment]
unordered_map <char, int> umap {{'I', 1}, {'V', 5}, {'X', 10}, {'L', 50}, {'C', 100}, {'D', 500}, {'M', 1000}};
string s;
int cevap = 0;

cin >> s;

int add = 0;
for (int i = 0; i < s.size(); i++) {
    if (umap[s[i]] < umap[s[i + 1]])
        cevap -= umap[s[i]];

    else
        cevap += umap[s[i]];
}

cout << cevap;
